src/utils/id.rs

```diff
@@ -0,0 +1,36 @@
+use std::str::FromStr;
+use std::{fs::read_dir, path::PathBuf};
+
+use anyhow::Result;
+
+pub fn get_maps_id(map_dir: PathBuf) -> Result<Vec<i32>> {
+    let maps_availables = read_dir(map_dir)?;
+    let mut maps_id = Vec::new();
+    const JAR_SUFFIX: &str = ".jar";
+    for raw_path in maps_availables {
+        match raw_path {
+            Ok(path) => {
+                let file_name = path.file_name();
+                match file_name.to_str() {
+                    Some(file_name) => {
+                        if file_name.ends_with(JAR_SUFFIX) {
+                            match file_name.strip_suffix(JAR_SUFFIX) {
+                                Some(map_id) => match i32::from_str(map_id) {
+                                    Ok(map_id) => {
+                                        maps_id.push(map_id);
+                                    }
+                                    Err(_e) => {}
+                                },
+                                None => {}
+                            }
+                        }
+                    }
+                    None => {}
+                };
+            }
+            Err(_e) => continue,
+        };
+    }
+    maps_id.sort();
+    return Ok(maps_id);
+}
```

Comment:
these pattern matches could all be removed by just using the `?` operator, or you could rewrite it with an iterator like this:
```suggestion
use std::fs;
use std::path::Path;

use anyhow::Result;
use itertools::Itertools;

pub fn get_maps_id(map_dir: impl AsRef<Path>) -> Result<Vec<i32>> {
    let mut maps_id: Vec<_> = fs::read_dir(map_dir)?
        .map_ok(|entry| entry.path())
        .filter_ok(|entry| entry.extension() == Some("jar".as_ref()))
        .filter_map_ok(|entry| Some(entry.file_stem()?.to_str()?.parse()))
        .flatten_ok()
        .try_collect()?;
    maps_id.sort();
    Ok(maps_id)
}
```

Reply:
also I'd rename this to `get_map_ids`

Reply:
Renamed and corrected the name of the function (I hadn't had the opportunity to manipulate itertools before and this allows me to do so, thanks for showing it to me!)